packages/snaps-utils/src/manifest/validation.ts

```diff
@@ -190,6 +190,34 @@ export type SnapPermissions = InferMatching<
   InitialPermissions
 >;
 
+const DynamicPermissionsStruct = refine(
+  PermissionsStruct,
+  'NonOverlappingKeys',
+  (value, context) => {
+    const { initialPermissions } = context.branch[0];
+
+    if (initialPermissions) {
+      const duplicateKeys = Object.keys(value).filter((key) =>
+        // eslint-disable-next-line no-prototype-builtins
+        initialPermissions.hasOwnProperty(key),
```

Comment:
Can we use `hasProperty` here?

Reply:
Refactored, using `hasProperty` from `utils` now.